futures-io/src/lib.rs

```diff
@@ -27,12 +27,16 @@ mod if_std {
 
     // Re-export some types from `std::io` so that users don't have to deal
     // with conflicts when `use`ing `futures::io` and `std::io`.
-    pub use self::StdIo::Error as Error;
-    pub use self::StdIo::ErrorKind as ErrorKind;
-    pub use self::StdIo::Result as Result;
-    pub use self::StdIo::IoSlice as IoSlice;
-    pub use self::StdIo::IoSliceMut as IoSliceMut;
-    pub use self::StdIo::SeekFrom as SeekFrom;
+    #[allow(clippy::useless_attribute)] // https://github.com/rust-lang/rust-clippy/issues/4106
```

Comment:
This false positive has already been fixed on the master of clippy (https://github.com/rust-lang/rust-clippy/pull/4107), but it may take some time for nightly to be released with this change.